File: basicTexture/app/src/main/java/basictexture/study/opengl/superbderrick/basictexture/MainActivity.java
package basictexture.study.opengl.superbderrick.basictexture;

import android.opengl.GLSurfaceView;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    private GLSurfaceView mGLSurfaceView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mGLSurfaceView = new GLSurfaceView(this);
        if (GLUtils.detectOpenGLES20(this))
        {
            // Tell the surface view we want to create an OpenGL ES 2.0-compatible
            // context, and set an OpenGL ES 2.0-compatible renderer.
            mGLSurfaceView.setEGLContextClientVersion(2);
            mGLSurfaceView.setRenderer(new SimpleTexture2DRenderer(this));
        }
        else
        {
            Log.e("SimpleTexture2D", "OpenGL ES 2.0 not supported on device.  Exiting...");
            finish();

        }
        setContentView(mGLSurfaceView);

    }

    @Override
    protected void onResume()
    {
        // Ideally a game should implement onResume() and onPause()
        // to take appropriate action when the activity looses focus
        super.onResume();
        mGLSurfaceView.onResume();
    }

    @Override
    protected void onPause()
    {
        // Ideally a game should implement onResume() and onPause()
        // to take appropriate action when the activity looses focus
        super.onPause();
        mGLSurfaceView.onPause();
    }

}

File: app/src/main/java/superbderrick/github/io/smile/MainActivity.kt
package superbderrick.github.io.smile

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 1
        happyButton.setOnClickListener({
            emotionalFaceView.happinessState = EmotionalFaceView.HAPPY


        })
// 2
        sadButton.setOnClickListener({
            emotionalFaceView.happinessState = EmotionalFaceView.SAD
        })
    }
}
